app/models/project.rb

```diff
@@ -1,3 +1,32 @@
+require 'csv'
+
 class Project < ApplicationRecord
 
+  def self.to_csv(ids = [])
+
+    csv = ''
+
+    project_columns = ["Number", "Project Title", "Donor(s)", "Status", "Start Date", "End Date",
+                       "Country", "Ocean-based Region", "Beneficiaries", "Implementing Agency",
+                       "Total Project Cost", "Co-funding entities", "Category", "Team Leader",
+                       "Further Information"]
+
+    csv << project_columns.join(',')
+    csv << "\r\n"
+
+    projects = Project.where(id: ids).order(start_date: :desc)
+
+    projects.to_a.each do |project|
+      project_attributes = project.attributes
+      project_attributes.delete("created_at")
+      project_attributes.delete("updated_at")
+      project_attributes = project_attributes.values.map{ |e| "\"#{e}\"" }
```

Comment:
I don't think you need this to generate a correct row. `join(',')` will return a string even if there are numbers in the array.

Reply:
@Levia I found that line was necessary because unfortunately some of the fields contain commas in their string which means that it breaks the CSV file. The only solution I found was to wrap everything in quotations like so:

`#project_attributes = project_attributes.values.map{ |e| "\"#{e}\"" }`

I just tried to remove that portion and the CSV is broken. However, there could be a better way that I haven't thought of.

Reply:
Ah I see. That should be fine than. Wen can test it out when we merge this with the frontend work.